packages/node_modules/@webex/plugin-meetings/src/meeting/audio.js

```diff
@@ -0,0 +1,101 @@
+import StateMachine from 'javascript-state-machine';
+import StateMachineHistory from 'javascript-state-machine/lib/history';
+
+import Media from '../media';
+import MeetingUtil from '../meeting/util';
+import {MEETING_AUDIO_STATE_MACHINE, EVENT_TRIGGERS, EVENT_TYPES} from '../constants';
+
+const handleTransition = (value, audio, meeting) => {
+  if (audio.mute && audio.self) {
+    value = MEETING_AUDIO_STATE_MACHINE.STATES.MUTE_SELF;
```

Comment:
value is `this.state`. Is it intentional to change that value in addition to returning it from the function? Totally may be, I just figured it would be one or the other.

Reply:
afaik, javascript does pass by value for strings? since this.state is a string type, value is a copy of the value stored in string. it then returns and alters state internal javascript-state-machine from to: function(...)
short summary, yes, it's intentional to get a copy of it, know what it is before it would've been changed, change it *if* it should be, and return the result to set what state should be, in some cases, there is no change, in which case no transition lifecycle events are triggered